lib/Sema/CSDiag.cpp

```diff
@@ -1333,7 +1333,7 @@ diagnoseTypeMemberOnInstanceLookup(Type baseObjTy,
 static DeclName
 findCorrectEnumCaseName(Type Ty, TypoCorrectionResults &corrections,
                         DeclName memberName) {
-  if (!memberName.isSimpleName())
+  if (memberName.isSpecial() || !memberName.isSimpleName())
```

Comment:
Instead of bailing out on special names, you can replace the `DeclBaseName::getIdentifier()` calls with `DeclBaseName::userFacingName()`.

Reply:
Right, but I don't think it makes sense in this particular case because enums can't have cases with special names.

Reply:
Maybe assert that its not special then :)

Reply:
I think assert might not be defensive enough because although enums can't have cases with special names it doesn't mean that nothing it going to attempt one and we'd probably get a crash with ambiguous stacktrace for that.